api/firmware/bluetooth.go

```diff
@@ -41,44 +41,42 @@ func (device *Device) queryBluetooth(request *messages.BluetoothRequest) (*messa
 
 // BluetoothUpgrade attempts an upgrade of the Bluetooth firmware.
 func (device *Device) BluetoothUpgrade(firmware []byte) error {
-	// Send initial upgrade request
-	initReq := &messages.BluetoothUpgradeInitRequest{
-		FirmwareLength: uint32(len(firmware)),
-	}
-	req := &messages.BluetoothRequest{
-		Request: &messages.BluetoothRequest_UpgradeInit{
-			UpgradeInit: initReq,
-		},
-	}
-
-	currentResponse, err := device.queryBluetooth(req)
-	if err != nil {
-		return err
-	}
+	return device.atomicQueries(func() error {
+		currentResponse, err := device.nonAtomicQueryBluetooth(&messages.BluetoothRequest{
+			Request: &messages.BluetoothRequest_UpgradeInit{
+				UpgradeInit: &messages.BluetoothUpgradeInitRequest{
+					FirmwareLength: uint32(len(firmware)),
```

Comment:
I might have missed this when we added the protobufs. The firmware length always fits in a u16. Is u32 correct here?

Reply:
Protobufs don't have a u16 type, just u32: https://protobuf.dev/programming-guides/proto3/#scalar